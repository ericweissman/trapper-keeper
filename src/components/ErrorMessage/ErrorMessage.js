import React from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'

const ErrorMessage = (props) => {
  const { error } = props
  return (
    <div className="error-msg">
      <h3 className='test'>there has been an error: {error}</h3>
      <Link to='/'>
        Return to All Notes
      </Link>
    </div>
  )
}

ErrorMessage.propTypes = {
  error: PropTypes.string,
}

ErrorMessage.defaultProps = {
  error: ''
}

export default ErrorMessage