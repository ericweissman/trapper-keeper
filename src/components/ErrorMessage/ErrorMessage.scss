.error-msg {
  text-align: center;
  font-family: $font-secondary;
  h3 {
    color: $primary-dark;
    text-transform: uppercase; 
  }
  a {
    text-decoration: underline;
    color: $primary-light
  }
  padding-top: 50px;
}