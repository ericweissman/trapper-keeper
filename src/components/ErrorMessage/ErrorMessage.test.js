import React from 'react'
import { shallow } from 'enzyme'
import ErrorMessage from './ErrorMessage'

describe('ErrorMessage', () => {
  let wrapper
  const mockError = '404'
  it('should match the correct snapshot', () => {
    wrapper = shallow(<ErrorMessage error={mockError}/>)
    expect(wrapper).toMatchSnapshot()
  })
})