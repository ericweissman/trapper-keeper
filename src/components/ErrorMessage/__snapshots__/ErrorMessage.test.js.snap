// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should match the correct snapshot 1`] = `
<div
  className="error-msg"
>
  <h3
    className="test"
  >
    there has been an error: 
    404
  </h3>
  <Link
    replace={false}
    to="/"
  >
    Return to All Notes
  </Link>
</div>
`;
